src/external.rs

```diff
@@ -100,9 +103,88 @@ pub trait External {
 	fn address(&mut self) -> Address {
 		unimplemented!()
 	}
+
+	fn as_any(&self) -> &Any;
+}
+
+#[derive(Clone, Default)]
+pub struct ExternalInstance {
+	pub storage: HashMap<H256, [u8; 32]>,
+	pub balances: HashMap<Address, U256>,
+	pub sender: Address,
+	pub value: U256,
+	pub address: Address,
+	pub origin: Address,
+	pub coinbase: Address,
+	pub difficulty: U256,
+	pub gas_limit: U256,
+	pub blocknumber: u64,
+	pub timestamp: u64,
 }
 
-/// Dummy unimplemeted external functions interface
-pub struct ExternalImpl;
+impl External for ExternalInstance {
+	fn storage_read(&mut self, key: &H256) -> Result<[u8; 32], Error> {
+		if let Some(value) = self.storage.get(key) {
+			Ok(value.clone())
+		} else {
+			Err(Error)
+		}
+	}
+
+	fn balance(&mut self, address: &Address) -> U256 {
+		self.balances[address]
+	}
+
+	fn storage_write(&mut self, key: &H256, value: &[u8; 32]) -> Result<(), Error> {
+		self.storage.insert(*key, value.clone());
+		Ok(())
+	}
+
+	fn sender(&mut self) -> Address {
+		self.sender
+	}
+
+	/// Invoked when contract is requesting coinbase extern
```

Comment:
These comments seems redudant to me, because they are repeating those in `External`.